For #21339: update styles for homescreen to match new specs

eliserichards opened the pull request:
For #21339

Updates the "show all" buttons on the home screen. Adds correct spacing in between jump back in cards.

### Pull Request checklist
<!-- Before submitting the PR, please address each item -->
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

eliserichards commented:
Failing UI tests: 
- verifyExpandedCollectionItemsTest
- createFirstCollectionTest

eliserichards commented:
@Mergifyio rebase

eliserichards commented:
UI tests failing again: 
* verifyExpandedCollectionItemsTest: [logs](https://console.firebase.google.com/u/0/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/7756624826495169639/executions/bs.9dc282bf38efa7aa/testcases/1/test-cases)
* createFirstCollectionTest: [logs](https://console.firebase.google.com/u/0/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/7756624826495169639/executions/bs.62b9ede231706736/testcases/1/test-cases)
* verifyAddTabButtonOfCollectionMenu: [logs](https://console.firebase.google.com/u/0/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/7756624826495169639/executions/bs.d2c8c8b11b5481e8/testcases/1/test-cases)
* renameCollectionTest: [logs](https://console.firebase.google.com/u/0/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/7756624826495169639/executions/bs.6f4a68c106a9e943/testcases/1/test-cases)

Adding ignores and filing a bug for help: #...

gabrielluong commented on app/src/main/java/org/mozilla/fenix/home/recenttabs/view/RecentTabs.kt:
```diff
@@ -116,6 +116,7 @@ private fun RecentTabItem(
             }
         }
     }
+    Spacer(modifier = Modifier.width(16.dp))
```
I think you are looking to increase the spacing between the header of "Recently saved " and not between each RecentTabItem (media, current tab). This change will in fact change increase the 8dp spacing to 24 (8+16) between each recent tab item (media, current open tab, search groups) that is provided by `verticalArrangement = Arrangement.spacedBy(8.dp)`. If you did want to increase the spacing between the recent tab and media to 24dp. You can change the spacedBy. 

I think the best way is to add a top margin of 16dp in the headers instead.

eliserichards replied:
This is modifying the area in between each recentTabItem, not the space between the headers. Should I add that `verticalArragment` to `Card` instead of the `Spacer`?

gabrielluong replied:
I am not seeing where we want this additional 16dp spacer in between the tabs. The current 8dp you see in the screenshot is handled by the `verticalArrangement = Arrangement.spacedBy(8.dp)` in the `Column` of `RecentTabs`. 

<img width="355" alt="Screen Shot 2021-09-22 at 3 23 30 PM" src="https://user-images.githubusercontent.com/1190888/134410565-93e872f6-c6c0-4167-a840-74cd060246e2.png">

eliserichards replied:
Ahh it was those pesky ios designs that have 16dp in between!

gabrielluong commented on app/src/main/res/layout/recent_bookmarks_header.xml:
```diff
@@ -12,31 +12,34 @@
     <com.google.android.material.button.MaterialButton
         android:id="@+id/showAllBookmarksButton"
-        style="@style/Button12TextStyle"
+        style="@style/Button15TextStyle"
```
I think you are looking at the iOS designs. The buttons for Android should be 14dp and purple in light theme.

eliserichards replied:
Duh. I was wondering why it was blue. Thanks for the catch 👍

gabrielluong commented on app/src/main/res/layout/history_metadata_header.xml:
```diff
@@ -6,29 +6,36 @@
+        android:paddingVertical="16dp"
+        android:paddingStart="16dp"
+        android:paddingEnd="0dp"
+        android:letterSpacing="-0.0024"
```
I think this is an artificial of looking at those iOS designs letterspacing spec. I don't think we have any for Android.

eliserichards replied:
Removing that as we speak ;)

gabrielluong commented on app/src/main/res/values-night/colors.xml:
```diff
@@ -93,7 +93,7 @@
 
     <!-- Home screen -->
-    <color name="home_show_all_button_text">@color/photonLightGrey50</color>
+    <color name="home_show_all_button_text">#AB71FF</color>
```
This should be equivalent to photonViolet40. You can usually see the color name in figma if you select the CSS.

eliserichards replied:
<img width="256" alt="image" src="https://user-images.githubusercontent.com/43795363/134415785-818db675-7ed2-4ef0-a935-74101a3a6341.png">

gabrielluong replied:
Oui, the color variable for photonViolet40 is what I am trying to say ab71ff
https://searchfox.org/mozilla-mobile/source/android-components/components/ui/colors/src/main/res/values/photon_colors.xml#180

eliserichards replied:
We don't have that in our .values folder on Fenix. What's the correct way to add a new photon color here?

gabrielluong replied:
Should simplify be `@color/photonViolet40`

eliserichards replied:
That doesn't work unless it's already defined in the .values. I'll use the create wizard and see if that works

eliserichards replied:
We'll have to add it to `mozilla.components.ui.colors` to be able to access it the correct way. I'll open a PR for that and we can keep #AB71FF for now

gabrielluong replied:
It's working fine for me locally. There's really no magic here. We were just using `@color/photonLightGrey50` before the changed to #AB71FF in the PR. 

```suggestion
    <color name="home_show_all_button_text">@color/photonViolet40</color>
```

gabrielluong replied:
It also already exists in ui-color https://searchfox.org/mozilla-mobile/source/android-components/components/ui/colors/src/main/res/values/photon_colors.xml#180

gabrielluong replied:
I am just blocking on this change since it should work. Otherwise, I suspect there might be some local build issue.

eliserichards replied:
that's in android components. Try it locally. We build our @color/photonXXX values based on https://github.com/mozilla-mobile/android-components/blob/releases/93.0/components/ui/colors/src/main/res/values/photon_colors.xml and you can see that violet40 isn't there

eliserichards replied:
<img width="573" alt="image" src="https://user-images.githubusercontent.com/43795363/134572857-f38bbdde-726e-4aec-a6c9-7b6790392023.png">

eliserichards replied:
I'm not adding that in this PR. I'm gonna expose it on AC then update here.

gabrielluong replied:
It's there in AC. https://github.com/mozilla-mobile/android-components/blob/main/components/ui/colors/src/main/res/values/photon_colors.xml#L180. We aren't marked to the 93 release in main.

Trust or ping me for troubleshooting your local dev.

gabrielluong replied:
We can also omit the change and I can add it in a separate PR if we are still having issues, but the color definitely exists in AC already.

gabrielluong commented on app/src/main/res/values/strings.xml:
```diff
@@ -118,11 +118,16 @@
     <string name="recent_tabs_show_all">Show all</string>
+    <!-- Content description for the button which navigates the user to show all recent tabs in the tabs tray. -->
+    <string name="recent_tabs_show_all_content_description">Show all recent tabs button</string>
+
```
nit: extra new line

gabrielluong reviewed (commented):
Will pull down and test it out after lunch

gabrielluong commented on app/src/main/res/layout/recent_tabs_header.xml:
```diff
@@ -6,29 +6,35 @@
     xmlns:android="http://schemas.android.com/apk/res/android"
```
There's something weird on this header compare to the others. Also, all the "Show all" button seems to have a weird constraint.

<img width="441" alt="Screen Shot 2021-09-22 at 6 12 50 PM" src="https://user-images.githubusercontent.com/1190888/134429061-eadfbf86-3762-405a-9631-7e50d47c43d7.png">

gabrielluong replied:
Another side effect of the current design is that the click gesture on the "Show all" shows that hard right edge compare to having it centered. Might be worthwhile to show it to @topotropic

gabrielluong replied:
This can be dealt with as a followup.

eliserichards commented:
@gabrielluong I cleaned everything up a bunch and removed all of the unnecessary white space around bookmarks and top sites, as well as making sure collections and "no collections" were matching. lmk what you think :)

eliserichards commented:
Making some changes to "customize" button based on Nicole's feedback...

eliserichards commented:
Modified the collections header so that it takes into account the top margins above each of the collections. Spaced out the "customize" button so that it matches the space in between each section:

<img width="323" src="https://user-images.githubusercontent.com/43795363/134576110-9cbf0979-b61b-4c00-9132-317ade862eec.png">

cadeyrn commented:
@eliserichards While you are at it, could you also fix the overlap of the card headers and the "show all" which happens in languages with longer translations? I checked out your current work on this issue and while it's better due to the smaller font it's still an issue. I attached a screenshot of the "recently saved" section so that you can see the issue (left: main branch, right: your branch for this issue).

![screenshot](https://user-images.githubusercontent.com/1100614/134703881-f2ec0a15-3625-451b-80bd-fd85396d24c0.png)

eliserichards commented:
@cadeyrn thank you so much for pointing that out! I'm going to land this and tackle that issue in a new PR :D

eliserichards commented:
Flaky test: 
* TitleHeaderBindingTest. `WHEN grouped tabs are added to the list THEN return false`

eliserichards commented:
@mergifyio rebase
